Functions/Example/Tests/FIRFunctionsTests.m

```diff
@@ -81,6 +83,29 @@ - (void)tearDown {
   [super tearDown];
 }
 
+- (void)testFunctionsInstanceIsStablePerApp {
+  FIROptions *options =
+      [[FIROptions alloc] initWithGoogleAppID:@"0:0000000000000:ios:0000000000000000"
+                                  GCMSenderID:@"00000000000000000-00000000000-000000000"];
+  [FIRApp configureWithOptions:options];
+
+  FIRFunctions *functions1 = [FIRFunctions functions];
+  FIRFunctions *functions2 = [FIRFunctions functionsForRegion:@"us-central"];
```

Comment:
Can you please comment that this is the default region and that's why we expect them to be the same?

Reply:
These are the same because the current implementation returns an existing FIRFunctions if one was already configured for the given app (in this case implicitly the default app), ignoring the region parameter. If we allow this and `functionsForCustomDomain:` to return new instances every time, they'll also have the issue of emulator settings not persisting across instances; otherwise we need to have a more complex cache key.